source/Missions/Services/Agents/Handlers/ShieldBreakHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Common.Messaging;
+using Common.Network;
+using Missions.Services.Agents.Messages;
+using Missions.Services.Network;
+using System;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Handlers
+{
+    public class ShieldBreakHandler
```

Comment:
Make an interface for this class and that interface should be a dependency of CoopArenaController

Also add this interface to the DI Module and make remove .AsSelf().InstancePerLifetimeScope().AutoActivate();